src/smefit/wcxf.py

```diff
@@ -0,0 +1,168 @@
+    "O8qu": {"wc": ["qu8_3311", "qu8_3322"]},
+    "O1qu": {"wc": ["qu1_3311", "qu1_3322"]},
+    "O8dt": {"wc": ["ud8_3311", "ud8_3322", "ud8_3333"]},
+    "O1dt": {"wc": ["ud1_3311", "ud1_3322", "ud1_3333"]},
+    "O8qd": {"wc": ["qd8_3311", "qd8_3322", "qd8_3333"]},
+    "O1qd": {"wc": ["qd1_3311", "qd1_3322", "qd1_3333"]},
+    # 4H quark operators
+    "OQQ1": {"wc": ["qq1_3333"], "value": [1.0 / 2.0]},
+    "OQQ8": {"wc": ["qq1_3333", "qq3_3333"], "value": [1.0 / 24.0, 1.0 / 8.0]},
+    "OQt1": {"wc": ["qu1_3333"]},
+    "OQt8": {"wc": ["qu8_3333"]},
+    "Ott1": {"wc": ["uu_3333"]},
+    # 4 leptons
+    "Oll": {"wc": ["ll_1221"]},
+    "Oll1111": {"wc": ["ll_1111"]},
+}
+
+# This creates a dictionary to go from Warsaw to SMEFiT.
+# In particular, given a point in the coefficient space of the Warsaw basis,
+# it tells you how to combine them to get the coefficients in the SMEFiT basis.
+# These are basically the equations between Wilson coefficients.
+# C_{SMEFiT} = sum_i coeff_i * C_{Warsaw, i}
+# Note that the flavour structure is assumed to hold and therefore the values
+# are inferred only from the 11 components.
+inverse_wcxf_translate = {
+    # Bosonic
+    "OWWW": {"wc": ["W"]},
+    "Opd": {"wc": ["phiBox"], "coeff": [-1.0]},
+    "OpD": {"wc": ["phiD"]},
+    "OpWB": {"wc": ["phiWB"]},
+    "OpG": {"wc": ["phiG"]},
+    "OpW": {"wc": ["phiW"]},
+    "OpB": {"wc": ["phiB"]},
+    "Op": {"wc": ["phi"]},
+    # Dipoles
+    "OtG": {"wc": ["uG_33"], "coeff": ["1/gs"]},
+    "OtW": {"wc": ["uW_33"], "coeff": [1.0]},
+    "OtZ": {"wc": ["uB_33", "uW_33"], "coeff": [-sw, cw]},
+    # Quark Currents
+    "O3pq": {"wc": ["phiq3_11"]},
+    "OpqMi": {"wc": ["phiq1_11", "phiq3_11"], "coeff": [1.0, -1.0]},
+    "O3pQ3": {"wc": ["phiq3_33"]},
+    "OpQM": {"wc": ["phiq1_33", "phiq3_33"], "coeff": [1.0, -1.0]},
+    "Opt": {"wc": ["phiu_33"]},
+    "Opui": {"wc": ["phiu_11"]},
+    "Opdi": {"wc": ["phid_11"]},
+    # Lepton Currents
+    "Opl1": {"wc": ["phil1_11"]},
+    "Opl2": {"wc": ["phil1_22"]},
+    "Opl3": {"wc": ["phil1_33"]},
+    "O3pl1": {"wc": ["phil3_11"]},
+    "O3pl2": {"wc": ["phil3_22"]},
+    "O3pl3": {"wc": ["phil3_33"]},
+    "Ope": {"wc": ["phie_11"]},
+    "Opmu": {"wc": ["phie_22"]},
+    "Opta": {"wc": ["phie_33"]},
+    # Yukawas
+    "Otp": {"wc": ["uphi_33"]},
+    "Ocp": {"wc": ["uphi_22"]},
+    "Obp": {"wc": ["dphi_33"]},
+    "Otap": {"wc": ["ephi_33"]},
+    # 2L2H quark operators
+    "O81qq": {"wc": ["qq1_1331", "qq3_1331"], "coeff": [1.0, 3.0]},
+    "O11qq": {
+        "wc": ["qq1_1133", "qq1_1331", "qq3_1331"],
+        "coeff": [1.0, 1.0 / 6.0, 1.0 / 2.0],
+    },
+    "O83qq": {
+        "wc": ["qq1_1331", "qq3_1331"],
+        "coeff": [1.0, -1.0],
+    },
+    "O13qq": {
+        "wc": ["qq3_1133", "qq1_1331", "qq3_1331"],
+        "coeff": [1.0, 1.0 / 6.0, -1.0 / 6.0],
+    },
+    "O8qt": {"wc": ["qu8_1133"]},
+    "O1qt": {"wc": ["qu1_1133"]},
+    "O8ut": {"wc": ["uu_1331"], "coeff": [2.0]},
```

Comment:
I have a question here about the flavour indices. In SMEFiT, we define c8ut as 2 * (uu_i33i) where i = 1,2, and we then argue that i = 1 or i = 2 are the same in our flavour assumptions (we don't single out the first and second generations). But this flavour symmetry is not respected by the anomalous dimension matrix in Wilson that governs the running, right? So are we sure writing uu_1331 is the same as writing uu_2332 here?

Reply:
The same question could be asked for the inverse transformation. I see they are consistent though.

Reply:
By the way, I have now checked the transformations in wcxf.py and agree with the transformations, only the above question remains :)

Reply:
Yeah, I guess we are assuming that the flavour assumptions are respected by the evolution... I don't see what else we can do, but if we use the Yukawa matrix that has only the top yukawa different from zero, I think the evolution shouldn't be able to distinguish the first 2 generations